observability/micrometer/src/main/java/org/wildfly/extension/micrometer/MicrometerDeploymentProcessor.java

```diff
@@ -21,32 +18,29 @@
 import org.jboss.as.server.deployment.DeploymentUnitProcessor;
 import org.jboss.as.weld.WeldCapability;
 import org.wildfly.extension.micrometer.api.MicrometerCdiExtension;
-import org.wildfly.extension.micrometer.registry.WildFlyRegistry;
+import org.wildfly.subsystem.service.ServiceInstaller;
 
 class MicrometerDeploymentProcessor implements DeploymentUnitProcessor {
-    private final boolean exposeAnySubsystem;
-    private final List<String> exposedSubsystems;
-    private final Supplier<WildFlyRegistry> registrySupplier;
+    static final String WELD_CAPABILITY_NAME = "org.wildfly.weld";
+
+    private final MicrometerSubsystemRegistrar.MicrometerDeploymentConfiguration config;
 
-    MicrometerDeploymentProcessor(boolean exposeAnySubsystem,
-                                  List<String> exposedSubsystems,
-                                  Supplier<WildFlyRegistry> registrySupplier) {
-        this.exposeAnySubsystem = exposeAnySubsystem;
-        this.exposedSubsystems = exposedSubsystems;
-        this.registrySupplier = registrySupplier;
+    MicrometerDeploymentProcessor(MicrometerSubsystemRegistrar.MicrometerDeploymentConfiguration config) {
+        this.config = config;
     }
 
     @Override
     public void deploy(DeploymentPhaseContext deploymentPhaseContext) throws DeploymentUnitProcessingException {
         final DeploymentUnit deploymentUnit = deploymentPhaseContext.getDeploymentUnit();
 
-        MicrometerDeploymentService.install(deploymentPhaseContext.getServiceTarget(),
-                deploymentPhaseContext,
-                deploymentUnit.getAttachment(DeploymentModelUtils.DEPLOYMENT_RESOURCE),
-                deploymentUnit.getAttachment(DeploymentModelUtils.MUTABLE_REGISTRATION_ATTACHMENT),
-                registrySupplier,
-                exposeAnySubsystem,
-                exposedSubsystems);
+        ServiceInstaller.builder(() -> new MicrometerDeploymentService(
```

Comment:
This doesn't look right.  The various ServiceInstaller.Builder factory methods generate generic Service implementations.  This particular method expects a Supplier of the value to be produced by the Service.
Presumably, this service produces a `MetricRegistration`, right?
So, perhaps you want something like this:

```java
Resource resource = deploymentUnit.getAttachment(DeploymentModelUtils.DEPLOYMENT_RESOURCE);
ManagementResourceRegistration registration = deploymentUnit.getAttachment(DeploymentModelUtils.MUTABLE_REGISTRATION_ATTACHMENT);
PathAddress address = ...;
Supplier<MetricRegistration> factory = () -> this.config.getCollectorSupplier().get().collectResourceMetrics(resource, registration, deploymentAddress::append, this.config.getSubsystemFilter());
ServiceInstaller.builder(factory)
    .onStop(MetricRegistration::unregister)
    .build()
    .install(deploymentPhaseContext);
```